server/broadcast.go

```diff
@@ -347,13 +359,30 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 	// send segment to the orchestrator
 	glog.V(common.DEBUG).Infof("Submitting segment nonce=%d manifestID=%s seqNo=%d orch=%s", nonce, cxn.mid, seg.SeqNo, sess.OrchestratorInfo.Transcoder)
 
-	res, err := SubmitSegment(sess, seg, nonce)
-	if err != nil || res == nil {
+	res, tErr := SubmitSegment(sess, seg, nonce)
+	// If TicketParams have expired, refresh them
+	if tErr != nil && tErr == pm.ErrTicketParamsExpired {
```

Comment:
Could we avoid some of this nesting and additional error checking if we called `sender.ValidateTicketParams()` instead? Something like:

```
err := sess.Sender.ValidateTicketParams(params)
if err != nil {
	if err != pm.ErrTicketParamsExpired {
		return nil, err
	}

	sess, err = refreshSession(sess)
	if err != nil {
		cxn.sessManager.removeSession(sess)
		return nil, err
	}
}

// Call SubmitSegment
```